pandas/tests/internals/test_internals.py

```diff
@@ -285,6 +285,12 @@ def test_delete(self):
         with pytest.raises(Exception):
             newb.delete(3)
 
+    def test_make_block_same_class(self):
+        # issue 19431
+        block = create_block('M8[ns, US/Eastern]', [3])
+        with pytest.warns(FutureWarning):
```

Comment:
use ``tm.assert_produces_warning`` instead. we catch this usage of ``pytest.warns`` in the linter and is not standard for pandas codease.

Reply:
good point.

Reply:
done